### ISSUE TITLE
Plugins/Modules/Addons?

### ISSUE THREAD

**Issue by AlexDaniel** [kensanata/oddmuse6#5]
I'd like to implement a bug tracker on top of oddmuse6, but what's the current situation with plugins? How should one go about implementing something like this?

**Comment by AlexDaniel** [comment]
Yeah, I think this is something that needs to be figured out early on… I see that different backends are supposed to be created using Oddmuse::Storage::Delegate, but all other modules are likely to use a different mechanism.

I understand that many things are frowned upon nowadays, but… you really can't beat the flexibility that was available in oddmuse5 where *anything* was changeable from a module or config file. How can we achieve the same here?

I was thinking about [`augment`](https://docs.perl6.org/syntax/augment), but then the docs say:

> For almost all situations, there are better solutions.

So what is it?

**Comment by kensanata** [comment]
“Help needed”

**Comment by AlexDaniel** [comment]
There's also [`wrap`](https://docs.perl6.org/routine/wrap) which is similar to what was used in oddmuse5.
